src/psyclone/tests/psyir/symbols/symbol_test.py

```diff
@@ -94,36 +92,93 @@ def test_symbol_initialisation():
-    is_argument and is_unresolved properties.
+    type. Also use this to test the is_automatic, is_import and
+    is_argument, is_unresolved, is_modulevar, is_static,
+    is_commonblock, is_unknown properties.
 
     '''
     symbol = Symbol('sym1')
-    assert symbol.is_local
+    assert symbol.is_automatic
     assert not symbol.is_import
     assert not symbol.is_argument
     assert not symbol.is_unresolved
+    assert not symbol.is_modulevar
+    assert not symbol.is_static
+    assert not symbol.is_commonblock
+    assert not symbol.is_unknown
 
     symbol.interface = ImportInterface(ContainerSymbol("my_mod"))
-    assert not symbol.is_local
+    assert not symbol.is_automatic
     assert symbol.is_import
     assert not symbol.is_argument
     assert not symbol.is_unresolved
+    assert not symbol.is_modulevar
+    assert not symbol.is_static
+    assert not symbol.is_commonblock
+    assert not symbol.is_unknown
 
     symbol.interface = ArgumentInterface()
-    assert not symbol.is_local
+    assert not symbol.is_automatic
     assert not symbol.is_import
     assert symbol.is_argument
     assert not symbol.is_unresolved
+    assert not symbol.is_modulevar
+    assert not symbol.is_static
+    assert not symbol.is_commonblock
+    assert not symbol.is_unknown
 
     symbol.interface = UnresolvedInterface()
-    assert not symbol.is_local
+    assert not symbol.is_automatic
     assert not symbol.is_import
     assert not symbol.is_argument
     assert symbol.is_unresolved
+    assert not symbol.is_modulevar
+    assert not symbol.is_static
+    assert not symbol.is_commonblock
+    assert not symbol.is_unknown
+
+    symbol.interface = DefaultModuleInterface()
+    assert not symbol.is_automatic
+    assert not symbol.is_import
+    assert not symbol.is_argument
+    assert not symbol.is_unresolved
+    assert symbol.is_modulevar
+    assert not symbol.is_static
+    assert not symbol.is_commonblock
+    assert not symbol.is_unknown
+
+    symbol.interface = StaticInterface()
+    assert not symbol.is_automatic
+    assert not symbol.is_import
+    assert not symbol.is_argument
+    assert not symbol.is_unresolved
+    assert not symbol.is_modulevar
+    assert symbol.is_static
+    assert not symbol.is_commonblock
+    assert not symbol.is_unknown
+
+    symbol.interface = CommonBlockInterface()
+    assert not symbol.is_automatic
+    assert not symbol.is_import
+    assert not symbol.is_argument
+    assert not symbol.is_unresolved
+    assert not symbol.is_modulevar
+    assert not symbol.is_static
+    assert symbol.is_commonblock
+    assert not symbol.is_unknown
+
+    symbol.interface = UnknownInterface()
+    assert not symbol.is_automatic
+    assert not symbol.is_import
+    assert not symbol.is_argument
+    assert not symbol.is_unresolved
+    assert not symbol.is_modulevar
+    assert not symbol.is_static
+    assert not symbol.is_commonblock
+    assert symbol.is_unknown
```

Comment:
I'm not sure about the name of this property: "is_unknown" is a bit vague.  However, I don't know what to suggest. Maybe just "is_unknown_interface"?

Reply:
I agree it could be confusing with the Unknown datatype. I changed it to the suggested is_unknown_interface